tests/unit/tools/mcp/test_unit_all_mcp.py

```diff
@@ -35,3 +35,38 @@ async def test_mcp_tools_loaded(
     await mcp_server._setup_tools(mcp_connection=mcp_connection)
 
     assert mcp_server.tools == list(tools)
+
+
+def test_filter_tools(
```

Comment:
Does  `test_sse_tool_filtering` overlap with this?
Also, why was not catching anything? is just a trap name to include `filtering` (I can't really tell what is going on in the test from GitHub UI because of the fixtures used)?

Reply:
That test was a trap, because after the fixture redesign, it was checking whether tools == tools (aka no filtering), it wasn't checking whether a tool was being filtered. That's a good call out: I'll fix that test. I'll keep it around because it does check the functionality at the _setup_tools() level (more than just the _filter_tools())  so I think it's nice to have both.